test/Microsoft.AspNet.Mvc.ViewFeatures.Test/Rendering/DefaultHtmlGeneratorTest.cs

```diff
@@ -60,6 +60,122 @@ public void GetCurrentValues_WithNullExpressionResult_ReturnsNull(bool allowMult
             Assert.Null(result);
         }
 
+        [Fact]
+        public void GetCurrentValues_WithNullExpression_Throws()
+        {
+            // Arrange
+            var metadataProvider = new TestModelMetadataProvider();
+            var htmlGenerator = GetGenerator(metadataProvider);
+            var viewContext = GetViewContext<Model>(model: null, metadataProvider: metadataProvider);
+            var modelExplorer = metadataProvider.GetModelExplorerForType(typeof(string), model: null);
+
+            var expected = string.Format(
+                            "The name of an HTML field cannot be null or empty. Instead " +
+                            "use methods {0}.{1} or {2}.{3} with a non-empty {4} " +
+                            "argument value.\r\nParameter name: expression",
+                            typeof(IHtmlHelper).FullName,
+                            nameof(IHtmlHelper.Editor),
+                            typeof(IHtmlHelper<>).FullName,
+                            nameof(IHtmlHelper<object>.EditorFor),
+                            "htmlFieldName");
+
+            // Act and assert
+            var ex = Assert.Throws<ArgumentException>(() => htmlGenerator.GetCurrentValues(
+                                            viewContext,
+                                            modelExplorer,
+                                            expression: null,
+                                            allowMultiple: true));
```

Comment:
Use the `Throws<T>` overload that has a `paramName` parameter and pass "expression".